### ISSUE TITLE
Numpy 2 build?

### ISSUE THREAD

**Issue by count0** [conda-forge/graph-tool-feedstock#150]
### Comment:

There was a bot a while back trying to automatically port the recipe to numpy 2 (#142), but upstream was not ready. It is ready now (since version 2.71) but it seems the bot has either given up or has been discontinued.

It's not clear if still something we need to do... If so, we should try.

@stuarteberg

**Comment by stuarteberg** [comment]
We need help from the @conda-forge/bot team.

On the [bot status page for the numpy-2 migration][1], `graph-tool` is listed under the "Not solvable" category.  It is showing errors like this:

```
linux_64_microarch_level1numpy2.0python3.12.____cpython: Cannot solve the request because of: _x86_64-microarch-level >=1 cannot be installed because there are no viable options:
└─ _x86_64-microarch-level 1 | 1 | 1 | 1 | 1 | 1 | 1 | 1 | 1 | 1 | 1 | 1 | 2 | 2 | 2 | 2 | 2 | 2 | 2 | 2 | 2 | 2 | 2 | 2 | 2 | 2 | 2 | 2 | 2 | 2 | 2 | 2 | 2 | 2 | 2 | 2 | 3 | 3 | 3 | 3 | 3 | 3 | 3 | 3 | 3 | 3 | 3 | 3 | 3 | 3 | 3 | 3 | 3 | 3 | 3 | 3 | 3 | 3 | 3 | 3 | 3 | 3 | 3 | 3 | 3 | 3 | 4 | 4 | 4 | 4 | 4 | 4 | 4 | 4 | 4 | 4 | 4 | 4 | 4 | 4 | 4 | 4 | 4 would require
   └─ __archspec ==1 core2, for which no candidates were found.
```

[1]: https://conda-forge.org/status/migration/numpy2


So it seems that somehow our enhancements for microarch-optimized builds are complicating things for the bot.  The bot was successfully triggered on May 29, before we merged our changes in #140, but was not able to run again after that.

It might be possible to just inspect the diff for #142 and manually apply those changes in a new PR, but it won't be practical to do that for all migrations in the future.

**Comment by beckermr** [comment]
Can you make an issue here in this repo https://github.com/regro/conda-forge-feedstock-check-solvable?

I suspect we need to add `__archspec` virtual packages to the solver testing logic.

**Comment by count0** [comment]
Closed via PR #152
